frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot-content-details.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import {
+  SectionHeading,
+  ResourceSummary,
+  ResourceLink,
+  navFactory,
+  Kebab,
+  humanizeBinaryBytes,
+} from '@console/internal/components/utils';
+import { VolumeSnapshotClassModel, VolumeSnapshotModel } from '@console/internal/models';
+import { referenceForModel, VolumeSnapshotContentKind } from '@console/internal/module/k8s';
+import { DetailsPage, DetailsPageProps } from '@console/internal/components/factory';
+import { Status } from '@console/shared';
+import { volumeSnapshotStatus } from '../../status';
+import { ResourceEventStream } from '@console/internal/components/events';
+
+const { editYaml, events } = navFactory;
+
+const Details: React.FC<DetailsProps> = ({ obj }) => {
+  const { deletionPolicy, driver } = obj?.spec;
+  const { volumeHandle, snapshotHandle } = obj?.spec?.source;
```

Comment:
```suggestion
  const { volumeHandle, snapshotHandle } = obj?.spec?.source || {};
```
Whenever you're using optional chaining with destructuring if the property doesn't exist then it will try to destructure `undefined` and fail.